py-polars/polars/internals/dataframe/frame.py

```diff
@@ -6668,23 +6669,17 @@ def rows(self, named: bool = False) -> list[tuple[Any, ...]] | list[Any]:
         ... )
         >>> df.rows()
         [(1, 2), (3, 4), (5, 6)]
-        >>> df.rows(named=True)
-        [Row(a=1, b=2), Row(a=3, b=4), Row(a=5, b=6)]
+        >>> df.rows(named=True)  # doctest: +IGNORE_RESULT
+        [{'a': 1, 'b': 2), {'a': 3, 'b': 4), {'a': 5, 'b': 6}]
 
         See Also
         --------
         iterrows : Row iterator over frame data (does not materialise all rows).
 
         """
         if named:
-            warnings.warn(
-                "Named rows will be changed from a namedtuple to a dictionary in the"
-                " next breaking release.",
-                category=FutureWarning,
-                stacklevel=2,
-            )
-            Row = namedtuple("Row", self.columns)  # type: ignore[misc]
-            return [Row(*row) for row in self._df.row_tuples()]
+            columns = self.columns
+            return [dict(zip(columns, row)) for row in self._df.row_tuples()]
```

Comment:
Not sure if storing `self.columns` in a local variable actually improves performance, or that Python will optimize this behind the scenes. I went with the safe route.

Reply:
No, you did the right thing - it's a small but real advantage; specifically the bytecode will be calling LOAD_FAST vs LOAD_ATTR inside the loop 👍

Reply:
Thanks for the explanation!

Reply:
If we're going to micro-optimise the hot loop, we may as well go all the way? :) 
```python
  dict_, zip_, columns = dict, zip, self.columns
  for row in rows_chunk:
      yield dict_(zip_(columns, row))
```
Now I admit that may _look_ silly, but it _will_ actually run marginally faster; the underlying bytecode will end up swapping out two LOAD_GLOBAL calls (for finding `zip` and `dict` in the global namespace) for two LOAD_FAST calls instead. It's free performance (albeit likely low single-digit), so why not? Can put a note next to it to explain the weirdness :)

Reply:
Free performance? Count me in. Will add this later.